torch/distributions/beta.py

```diff
@@ -0,0 +1,44 @@
+from numbers import Number
+
+import torch
+from torch.autograd import Variable
+from torch.distributions.dirichlet import Dirichlet
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+
+class Beta(Distribution):
+    r"""
+    Creates a Beta distribution parameterized by concentration `alpha` and `beta`.
+
+    Example::
+
+        >>> m = Beta(torch.Tensor([0.5]), torch.Tensor([0.5]))
+        >>> m.sample()  # Beta distributed with concentrarion alpha
+         0.1046
+        [torch.FloatTensor of size 2]
+
+    Args:
+        alpha (Tensor or Variable): concentration parameter of the distribution
+    """
+    has_rsample = True
+
+    def __init__(self, alpha, beta):
+        if isinstance(alpha, Number) and isinstance(beta, Number):
+            alpha_beta = torch.Tensor([alpha, beta])
+        else:
+            alpha, beta = broadcast_all(alpha, beta)
+            alpha_beta = torch.stack([alpha, beta], -1)
```

Comment:
Is this special handling for scalars expected? I thought we decided to just treat them as 1d arrays everywhere for consistency. Not a problem for merging, just wanted to make sure it's not like this by mistake.

Reply:
This is intended to match the behavior of other distributions. This special handling will no longer be needed once we have scalar support; until then you can get scalar-like shapes by passing `float`s as parameters.